Report more metrics and add output annotations

bibilujan opened the pull request:
We want to report more metrics after ichorCNA is run, previously only coverage was reported. https://jira.oicr.on.ca/browse/GRD-422 , we now want to report: 

-   mean coverage
-   total reads 
-   lanes sequenced
-   reads per lane

In addition, there is a request to sort the output pdfs to folders for easier review of the data: https://jira.oicr.on.ca/browse/GBS-3418

For this we are adding annotations to the pdf output files.

mlaszloffy commented on ichorCNA.wdl:
```diff
@@ -370,6 +461,50 @@ task runIchorCNA {
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
   command <<<
```
Can `set -euo pipefail` be added here?

mlaszloffy commented on ichorCNA.wdl:
```diff
@@ -483,7 +596,179 @@ task runIchorCNA {
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
+  command <<<
```
Can `set -euo pipefail` be added here?

mlaszloffy commented on ichorCNA.wdl:
```diff
@@ -483,7 +596,179 @@ task runIchorCNA {
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
+  command <<<
```
Can `set -euo pipefail` be added here?

bibilujan commented:
@mlaszloffy the build was failing because there was a pipefail at the read_count calculation, not a "real" failure, but based on what I could understand maybe something similar to this https://stackoverflow.com/questions/41516177/bash-zcat-head-causes-pipefail where head get the output before the first command is done. Changed it to grep and get the same result and no pipefail. 

A good catch was that read_counts was being calculated differently on premerge bams and post, so I have updated this to the new command.
